### PR TITLE
fix: validate mercurius is already registered

### PR DESCRIPTION
**Pull request by TomasTM**
@simone-sanfratello as per Simone's instructions:

> we should use fastify's plugin mechanism to do this, exposed by fastify-plugin.

I made this quick change. please let me know your thoughts

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,6 +46,9 @@ module.exports = fp(async function (app, { all, policy, ttl, cacheSize, skip, st
       cacheSize
     })
   }
+}, {
+  fastify: '>=3.x',
```
Comment (mcollina) [inline index.js:50]:
```suggestion
  fastify: '^3.x',
```

Comment (simoneb) [reply index.js:50]: in auth it's been done with the >= specifier. any reason why we should do it one way or the other?

https://github.com/mercurius-js/auth/blob/main/index.js#L32

**Comment by simone-sanfratello** [comment]
lgtm

**Comment by simoneb** [comment]
@TomasTM let's link this PR to the [original issue](https://github.com/mercurius-js/cache/issues/27) and include the other changes as well please.

File: index.js (hunk 1 above)
Comment (mcollina) [reply index.js:50]: Using the caret is better as it will not support Fastify v4 when it come out without manual change.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**